psi4/src/psi4/libmints/writer.cc

```diff
@@ -636,15 +636,19 @@ void FCHKWriter::write(const std::string &filename) {
 NBOWriter::NBOWriter(std::shared_ptr<Wavefunction> wavefunction) : wavefunction_(wavefunction) {}
 
 void NBOWriter::write(const std::string &filename) {
-    int pure_order[][7] = {
-        {1, 0, 0, 0, 0, 0, 0},        // s
-        {103, 101, 102, 0, 0, 0, 0},  // p
+    int pure_order[][11] = {
+        {1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},        // s
+        {103, 101, 102, 0, 0, 0, 0, 0, 0, 0, 0},  // p
         // z2  xz   yz  x2-y2 xy
-        {255, 252, 253, 254, 251, 0, 0},  // d
+        {255, 252, 253, 254, 251, 0, 0, 0, 0, 0, 0},  // d
         // z(z2-r2), x(z2-r2), y(z2-r2) z(x2-y2), xyz, x(x2-y2), y(x2-y2)
-        {351, 352, 353, 354, 355, 356, 357}  // f
+        {351, 352, 353, 354, 355, 356, 357, 0, 0, 0, 0},  // f
+        {451, 452, 453, 454, 455, 456, 457, 458, 459, 0, 0},  // g
+        {551, 552, 553, 554, 555, 556, 557, 558, 559, 560, 561}  // h
```

Comment:
Switch to ```std::vector<int>```?

Reply:
Python spoiled me, so my C++ typing is weak, but wouldn't it be`const std::vector<std::vector<int> >`? Or am I misunderstanding C++ typing? Either way, I agree that the zeros are avoidable ugliness.

Reply:
Right, this is a two-dimensional array, so yes: ```std::vector< std::vector<int> >```. There's also the `pure_order` array that should be changed.

Similarly, you could change the 
```
map = new int[nmo];
auto *skip = new bool[nmo];
```
to using `std::vector` to avoid manual delete statements.

Reply:
I'm fine with the std::vector switch, but changes to MOWriter are outside the scope of this PR.